src/MSBuild/MSBuild.csproj

```diff
@@ -261,7 +261,8 @@
        the bin folder. -->
   <Target Name="AddRefAssemblies"
           DependsOnTargets="ResolveAssemblyReferences"
-          BeforeTargets="AssignTargetPaths">
+          BeforeTargets="AssignTargetPaths"
+          Condition="'$(MonoBuild)' != 'true'">
```

Comment:
I'm surprised to see this difference. Can you go deeper on what's up?

Reply:
This is required because we want to instead use mono's assemblies.